cognite/neat/graph/stores/_base.py

```diff
@@ -1,291 +1,165 @@
+            postAsEncoded=False,
+            autocommit=False,
+        )
+        graph = Graph(store=store)
+        return cls(graph, rules)
+
+    @classmethod
+    def from_oxi_store(cls, storage_dir: Path | None = None, rules: InformationRules | None = None) -> "Self":
+        """Creates a NeatGraphStore from an Oxigraph store."""
+        local_import("pyoxigraph", "oxi")
+        import pyoxigraph
+
+        from cognite.neat.graph.stores._oxrdflib import OxigraphStore
+
+        # Adding support for both oxigraph in-memory and file-based storage
+        for i in range(4):
+            try:
+                oxi_store = pyoxigraph.Store(path=str(storage_dir) if storage_dir else None)
+                break
+            except OSError as e:
+                if "lock" in str(e) and i < 3:
+                    continue
+                raise e
+        else:
+            raise Exception("Error initializing Oxigraph store")
 
-    def query(self, query: str) -> Result:
-        """Returns the result of the query."""
-        start_time = time.perf_counter()
-        result = self.graph.query(query)
-        stop_time = time.perf_counter()
-        elapsed_time = stop_time - start_time
-        prom_qsm.labels("query").observe(elapsed_time)
-        prom_sq.labels("query").set(elapsed_time)
-        return result
+        graph = Graph(store=OxigraphStore(store=oxi_store))
+        graph.default_union = True
 
-    def serialize(self, *args, **kwargs):
-        """Serializes the graph."""
-        return self.graph.serialize(*args, **kwargs)
+        return cls(graph, rules)
 
-    def query_delayed(self, query) -> Iterable[Triple]:
-        """Returns the result of the query, but does not execute it immediately.
+    def write(self, extractor: TripleExtractors) -> None:
+        if isinstance(extractor, RdfFileExtractor):
+            self._parse_file(extractor.filepath, extractor.mime_type, extractor.base_uri)
+        else:
+            self._add_triples(extractor.extract())
 
-        The query is not executed until the result is iterated over.
+    def _parse_file(
+        self,
+        filepath: Path,
+        mime_type: MIMETypes = "application/rdf+xml",
+        base_uri: URIRef | None = None,
+    ) -> None:
+        """Imports graph data from file.
 
         Args:
-            query: SPARQL query to execute
-
-        Returns:
-            An iterable of triples
-
+            filepath : File path to file containing graph data, by default None
+            mime_type : MIME type of graph data, by default "application/rdf+xml"
+            add_base_iri : Add base IRI to graph, by default True
         """
-        return _DelayedQuery(self.graph, query)
 
-    @abstractmethod
-    def drop(self) -> None:
-        """Drops the graph."""
-        raise NotImplementedError()
+        # Oxigraph store, do not want to type hint this as it is an optional dependency
+        if type(self.graph.store).__name__ == "OxigraphStore":
```

Comment:
I guess this is the trade-off for not base classing. Worth it :)

Reply:
Unft. yest